src/language-js/flow-comments.js

```diff
@@ -0,0 +1,15 @@
+"use strict";
+
+function hasFlowShorthandAnnotationComment(node) {
+  // https://flow.org/en/docs/types/comments/
+  // Syntax example: const r = new (window.Request /*: Class<Request> */)("");
+
+  return (
+    node.extra &&
+    node.extra.parenthesized &&
+    !!node.trailingComments &&
+    node.trailingComments[0].value.startsWith(":")
```

Comment:
Can the comment be formatted as `/* : foo */`? If so, perhaps use `.value.trimLeft()`?

Reply:
Good point, it seems whitespace between `/*` and `:` is indeed allowed.